### PR TITLE
[core.stdc.math] everything is pure

### PR DESCRIPTION
**Pull request by thewilsonator**
Everything in core.stdc.math so use an attribute at the top.

### CODE REVIEW

File: src/core/stdc/math.d [hunk 1]
```diff
@@ -671,7 +672,7 @@ else version (CRuntime_Musl)
         FP_FAST_FMAL = 0,
     }
 
-  pure {
+  {
```
Comment (Geod24) [inline src/core/stdc/math.d:675]: Error here

**Comment by thewilsonator** [comment]
Thanks, fixed.

**Comment by ibuclaw** [comment]
I don't think you can say that everything is pure when most alter errno.

**Comment by Geod24** [comment]
For ref: https://en.cppreference.com/w/cpp/numeric/math/math_errhandling

**Comment by ibuclaw** [comment]
For your reference:
```
echo "module gcc.builtins;" | gdc -fsyntax-only -x d -Hf /dev/stdout - | less
```
Functions that are not intrinsically pure (because they store into memory).
- frexp{f,l}
- gamma{f,l}
- gamma_r{f,l}
- lgamma{f,l}
- lgamma_r{f,l}
- modf{f,l}
- remquo{f,l}
- sincos{f,l}

**Comment by n8sh** [comment]
I think https://github.com/dlang/druntime/pull/2045 doesn't leave anything in `core.stdc.math` that can be marked `pure` unless the definition of `pure` is changed to allow changes to `errno`. IIRC Walter Bright has toyed with that idea in the past.

**Comment by Geod24** [comment]
> unless the definition of `pure` is changed to allow changes to `errno`

I really hope we don't go in that direction. Logical `pure` would be as useful as logical `const`.

**Comment by ibuclaw** [comment]
> > unless the definition of `pure` is changed to allow changes to `errno`
> 
> I really hope we don't go in that direction. Logical `pure` would be as useful as logical `const`.

Well pure+errno is the main reason why we do things ourselves in std.math over just forwarding to libm.

**Comment by RazvanN7** [comment]
Closing as per:

>I think #2045 doesn't leave anything in core.stdc.math that can be marked pure unless the definition of pure is changed to allow changes to errno. IIRC Walter Bright has toyed with that idea in the past.

**Pull request closed without merging**